### PR TITLE
BEI-40: Update a recipe

### PR DESCRIPTION
**Pull request by LucianaDera**
WHY THIS PR?
We want users to have the ability to update created recipes.

WHAT DOES THIS PR DO?
It allows authenticated users update an existing recipe.

THINGS YOU MAY WANT THE REVIEWER TO NOTE.
N/A

### CODE REVIEW

File: src/core/controllers/recipe.ts [hunk 1]
```diff
@@ -107,3 +107,46 @@ export const createRecipe = async (
...
+    const updateOptions = { where: { slug: recipeSlug } };
+    const updatedRecipeData = {
+      description,
+      title,
+      servings,
+      image,
+      duration_in_minutes,
+      user_id: user.id,
+      slug: recipe.slug
+    };
```
Comment (RICHYDOS) [inline src/core/controllers/recipe.ts:144]: Nice work. The only issue I have is that this doesn't return a full recipe. It only returns the recipe table. Check the documentation for what I mean

**Review by RICHYDOS — CHANGES_REQUESTED** [review]
Nice work.

File: src/core/controllers/recipe.ts (hunk 1 above)
Comment (hizikex) [reply src/core/controllers/recipe.ts:144]: I agree with this

**Review by hizikex — CHANGES_REQUESTED** [review]

File: src/core/controllers/recipe.ts [hunk 2]
```diff
@@ -107,3 +107,46 @@ export const createRecipe = async (
...
+      duration_in_minutes
+    } = req.body;
+    const user = res.locals.user;
+    if(!user){
+      throw new AuthorizationError(
+        "Unauthorized",
+        new Error("not_authenticated"),
+        {}
+      );
+    }
```
Comment (LordUche) [inline src/core/controllers/recipe.ts:128]: This is not necessary

**Review by LordUche — CHANGES_REQUESTED** [review]

File: src/core/controllers/recipe.ts [hunk 3]
```diff
@@ -107,3 +109,50 @@ export const createRecipe = async (
...
+
+    const updatedRecipeData = {
+      description,
+      title,
+      servings,
+      image,
+      duration_in_minutes,
+      user_id: user.id,
+      slug: `${newSlugBase}-${uniqueId}`,
+    };
```
Comment (RICHYDOS) [inline src/core/controllers/recipe.ts:147]: What if a user doesn't update the title? Is a new slug generated or nothing happens?

**Review by RICHYDOS — APPROVED** [review]

**Review by hizikex — APPROVED** [review]

File: logs/combined.log
Comment (LordUche) [inline logs/combined.log:1]: Please ignore log files

File: src/core/controllers/recipe.ts [hunk 4]
```diff
@@ -107,3 +109,59 @@ export const createRecipe = async (
...
+
+    const updatedRecipeData= {
+      user_id: recipe.user_id,
+      description: recipe.description,
+      image: recipe.image,
+      servings: recipe.servings,
+      duration_in_minutes: recipe.duration_in_minutes,
+      title: title, 
+      slug:`${newSlugBase}-${numericId}`
+    };
```
Comment (LordUche) [inline src/core/controllers/recipe.ts:149]: All these can be done in a `beforeUpdate` hook

File: src/core/controllers/recipe.ts [hunk 5]
```diff
@@ -107,3 +109,59 @@ export const createRecipe = async (
...
   }
 };
+
+export const updateRecipe = async (
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => {
+  try {
+    const { title, } = req.body;
```
Comment (LordUche) [inline src/core/controllers/recipe.ts:119]:
```suggestion
    const { title, description } = req.body;
```

**Review by LordUche — CHANGES_REQUESTED** [review]

**Review by LordUche — APPROVED** [review]

**Pull request merged**